gpg availability is not properly checked, which results in four
tests failing on Solaris where it is not available.

Can be offered upstream.

--- dulwich-0.20.42/dulwich/tests/test_porcelain.py
+++ dulwich-0.20.42/dulwich/tests/test_porcelain.py
@@ -57,6 +57,11 @@ from dulwich.tests.utils import (
     make_object,
 )
 
+try:
+    import gpg
+except ImportError:
+    gpg = None
+
 
 def flat_walk_dir(dir_to_walk):
     for dirpath, _, filenames in os.walk(dir_to_walk):
@@ -85,6 +90,7 @@ class PorcelainTestCase(TestCase):
         self.assertLess(time.time() - ts, 50)
 
 
+@skipIf(gpg is None, "gpg is not available")
 class PorcelainGpgTestCase(PorcelainTestCase):
     DEFAULT_KEY = """
 -----BEGIN PGP PRIVATE KEY BLOCK-----
@@ -1119,8 +1125,6 @@ class RevListTests(PorcelainTestCase):
 class TagCreateSignTests(PorcelainGpgTestCase):
 
     def test_default_key(self):
-        import gpg
-
         c1, c2, c3 = build_commit_graph(
             self.repo.object_store, [[1], [2, 1], [3, 1, 2]]
         )
